messaging/src/main/java/org/axonframework/eventhandling/processors/streaming/pooled/MergeTask.java

```diff
@@ -135,11 +135,14 @@ private CompletableFuture<TrackingToken> tokenFor(int segmentId) {
     }
 
     private CompletableFuture<TrackingToken> fetchTokenInUnitOfWork(int segmentId) {
-        return unitOfWorkFactory
-                .create()
-                .executeWithResult(context ->
-                                           CompletableFuture.completedFuture(tokenStore.fetchToken(name, segmentId))
-                );
+        return joinAndUnwrap(unitOfWorkFactory
+                                     .create()
+                                     .executeWithResult(context ->
+                                                                CompletableFuture.completedFuture(
+                                                                        tokenStore.fetchToken(name,
+                                                                                              segmentId,
+                                                                                              null))
```

Comment:
```suggestion
                                                                                              context))
```

We should pass the `ProcessingContext` here.